src/components/slider/slider.tsx

```diff
@@ -124,16 +124,49 @@ export class Slider {
             return;
         }
 
+        const inputElement: HTMLInputElement = element.querySelector('input');
+        const value = this.getValue();
+
         /*
         For some reason the input element's `value` attribute is removed
         (probably by Stencil) when the element is first rendered. But if the
         attribute is missing when MDCSlider is initialized (MDC v11.0.0),
         MDCSlider crashes.
         So we add the attribute right before initializing MDCSlider. /Ads
         */
-        element
-            .querySelector('input')
-            .setAttribute('value', `${this.multiplyByFactor(this.getValue())}`);
+        inputElement.setAttribute('value', `${this.multiplyByFactor(value)}`);
+
+        /*
+        When creating the `mdcSlider` component, its important that the value set in
+        the input field obeys the range and the step size.
+
+        The MDCSlider will throw an exception unless the value in the input element
+        is dividible by the step value and is in the provided range.
+        If an exception occurs, this component will crash and it will be impossible to change
+        its value.
+        The logic below ensures that the component will render even though the
+        provided value is wrong.
+        This could be considered wrong, but it at least fixes so that it's possible
+        to change the value from the UI.
+        */
+        const greaterThanOrEqualMin = value >= this.valuemin;
+        const lessThanOrEqualMax = value <= this.valuemax;
+        const dividableByStep = value % this.step === 0;
+        const delta = 0.01;
+
+        if (!greaterThanOrEqualMin) {
+            const newMin = this.multiplyByFactor(value) - delta;
+            inputElement.setAttribute('min', `${newMin}`);
```

Comment:
This seems like a case where we really would benefit from having some tests. If you want to avoid testing the whole component, breaking out just the calculations to its own function and testing that might be a good idea. I learnt a trick for testing "private" functions on classes recently, which might be useful:

```ts
const testMe = new SomeClass(); // I think we could probably either get hold of the object for an element, or perhaps simply import `Slider` and do `new Slider()`?
const result = (testMe as any).privateFunctionToTest();
```

If that doesn't work, we can also move the function to a separate file, and import it. Then it can be imported and tested from that file.

I'm trying to prioritise the inline-create review, but if you need help getting started with a test file, hit me up 😊

Reply:
Aha, nice to know!

I think I would rather create a new file/module though where I could expose those functions as public functions instead and then import them in the slider component.

@Befkadu1 - I would say that we don't need to document that since we're documenting that they should set values that is in the range. Other scenarios should be considered as undefined behavior IMHO. We have just fixed such that the component no longer crashes if provided an empty value.